File: Week 2 Arrays/Problem set 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");
    float l = 0;
    float w = 1;
    float s = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            l++;
        }

        if (text[i] == 32)
        {
            w++;
        }

        if (text[i] == 46 || text[i] == 63 || text[i] == 33)
        {
            s++;
        }
    }

    float L = (l / w) * 100;
    float S = (s / w) * 100;

    int grade = round(0.0588 * L - 0.296 * S - 15.8);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: Week 6 Python/Problem set 6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
l = 0
w = 1
s = 0
for char in text:
    if char.isalpha():
        l += 1
    elif char == ' ':
        w += 1
    elif char in ['.', '!', '?']:
        s += 1
grade = round(0.0588 * ((l / w) * 100) - 0.296 * ((s / w) * 100) - 15.8)
if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")
